src/MSBuild/TerminalLogger/Project.cs

```diff
@@ -69,4 +73,17 @@ public void AddBuildMessage(MessageSeverity severity, string message)
         BuildMessages ??= new List<BuildMessage>();
         BuildMessages.Add(new BuildMessage(severity, message));
     }
+
+    /// <summary>
+    /// Filters the build messages to only include errors and warnings.
+    /// </summary>
+    /// <returns>A sequence of error and warning build messages.</returns>
+    public IEnumerable<BuildMessage> GetBuildErrorAndWarningMessages()
```

Comment:
How many of these are here please? This looks like potentially iterating over a huge collection.

Reply:
This iterates over all messages, right? That is where my concern comes from.

Reply:
Specifically this
```
BuildMessages.Where(message =>
                message.Severity == MessageSeverity.Error ||
                message.Severity == MessageSeverity.Warning);
```
Which looks like it goes all messages ever emitted - even for the subsequent .Any() call, which checks if the returned collection is non-empty.

Reply:
I shared my thought in the different thread.